src/renderer/components/opds/PageNavigation.tsx

```diff
@@ -0,0 +1,131 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import * as React from "react";
+
+import { TranslatorProps, withTranslator } from "readium-desktop/renderer/components/utils/hoc/translator";
+
+import * as ArrowRightIcon from "readium-desktop/renderer/assets/icons/baseline-arrow_forward_ios-24px.svg";
+import * as ArrowLeftIcon from "readium-desktop/renderer/assets/icons/baseline-arrow_left_ios-24px.svg";
+import SVG from "readium-desktop/renderer/components/utils/SVG";
+
+import { OpdsResultPageInfos, OpdsResultUrls } from "readium-desktop/common/views/opds";
+
+import * as styles from "readium-desktop/renderer/assets/styles/opds.css";
+
+interface Props extends TranslatorProps {
+    goto: (url: string) => void;
+    urls: OpdsResultUrls;
+    page: OpdsResultPageInfos;
+}
+
+class PageNavigation extends React.Component<Props> {
+    private currentPage: number = 1;
```

Comment:
Ah, *strictly-speaking* starting at `1` is not correct. *In practice* though, this "works" (read below for explanation).

OPDS feed / Atom "pagination" metadata only provides `numberOfItems` and `itemsPerPage`, but not "currentPage". In other words, the pagination context is incomplete, so we have to infer it in order to display meaningful information to the end-user ... but this is not 100% reliable.

PS: also note that the URLs of link `rel` = `previous` or `next` (and also Atom's `first` or `last`) are not *templated* URLs (at least not in OPDS < v2). So, for example the Feedbooks `?page=2` query parameter is totally arbitrary, it could be `/page/2`, etc.)

Consequently, it is possible (right now, in Thorium) to import an OPDS feed that is *not* at "page" `1`. For example:
https://catalog.feedbooks.com/publicdomain/browse/en/top.atom?cat=FBFIC009000&page=2
=>
http://readium2.herokuapp.com/opds-v1-v2-convert/https%3A%2F%2Fcatalog.feedbooks.com%2Fpublicdomain%2Fbrowse%2Fen%2Ftop.atom%3Fcat%3DFBFIC009000%26page%3D2

...resulting in incorrect "currentPage" information displayed to the user (see screenshot).

So, although *strictly-speaking* the current implementation of "curentPage" (i.e. always starts at `1`) is incorrect, like I said before, *in practice* this is an acceptable hack for a boundary condition that is unlikely to occur, because we can expect users *not* to import OPDS feeds part-way through a pagination sequence.

Thoughts?

![ThoriumOPDSPagination](https://user-images.githubusercontent.com/625201/65784382-d261d580-e149-11e9-921b-4331318c78d4.png)

Reply:
This is a "no fix", as explained above (in the concluding paragraph)